src/common/utils/create-router-context/index.ts

```diff
@@ -0,0 +1,19 @@
+import { DEFAULT_MATCH } from '../../constants';
+import { MatchParams, Query, Route } from '../../types';
+import matchRoute from '../match-route';
+import generatePath from '../generate-path';
+
+export const createRouterContext = (
+  route: Route,
+  params: MatchParams = {},
```

Comment:
So the gist is I'd be able to pass in a custom key here to store different sets of data for the same resource + route?

Reply:
Well, we still go through `getKey` in the resource, as the url parameters are also used by `getData`. If we just specify the key, we don't know how to call `refresh` for instance, as `getData` needs the same router context.
So instead we require all the bits that might be needed by  `getKey` and `getData` so we can safely operate on the resource without being worried